dspace-api/src/main/java/org/dspace/discovery/SolrServiceResourceRestrictionPlugin.java

```diff
@@ -54,34 +55,44 @@
     protected ResourcePolicyService resourcePolicyService;
 
     @Override
-    public void additionalIndex(Context context, DSpaceObject dso, SolrInputDocument document) {
-        try {
-            List<ResourcePolicy> policies = authorizeService.getPoliciesActionFilter(context, dso, Constants.READ);
-            for (ResourcePolicy resourcePolicy : policies) {
-                String fieldValue;
-                if (resourcePolicy.getGroup() != null) {
-                    //We have a group add it to the value
-                    fieldValue = "g" + resourcePolicy.getGroup().getID();
-                } else {
-                    //We have an eperson add it to the value
-                    fieldValue = "e" + resourcePolicy.getEPerson().getID();
+    public void additionalIndex(Context context, IndexableObject idxObj, SolrInputDocument document) {
+        if (idxObj instanceof DSpaceObject) {
+            DSpaceObject dso = (DSpaceObject) idxObj;
+            try {
+                List<ResourcePolicy> policies = authorizeService.getPoliciesActionFilter(context, dso, Constants.READ);
+                for (ResourcePolicy resourcePolicy : policies) {
+                    String fieldValue;
+                    if (resourcePolicy.getGroup() != null) {
+                        //We have a group add it to the value
+                        fieldValue = "g" + resourcePolicy.getGroup().getID();
+                    } else {
+                        //We have an eperson add it to the value
+                        fieldValue = "e" + resourcePolicy.getEPerson().getID();
 
-                }
+                    }
 
-                document.addField("read", fieldValue);
+                    document.addField("read", fieldValue);
 
-                //remove the policy from the cache to save memory
-                context.uncacheEntity(resourcePolicy);
+                    //remove the policy from the cache to save memory
+                    context.uncacheEntity(resourcePolicy);
+                }
+            } catch (SQLException e) {
+                log.error(LogManager.getHeader(context, "Error while indexing resource policies",
+                                               "DSpace object: (id " + dso.getID() + " type " + dso.getType() + ")"));
             }
-        } catch (SQLException e) {
-            log.error(LogManager.getHeader(context, "Error while indexing resource policies",
-                                           "DSpace object: (id " + dso.getID() + " type " + dso.getType() + ")"));
         }
     }
 
     @Override
     public void additionalSearchParameters(Context context, DiscoverQuery discoveryQuery, SolrQuery solrQuery) {
         try {
+            // skip workspace and workflow queries as security for it them is builtin in the SolrServiceImpl
```

Comment:
As mentioned above, I would prefer to have the security for workspace and workflow queries also in a dedicated plugin.

Reply:
see https://github.com/DSpace/DSpace/pull/2391#discussion_r272121467